packages/uniforms-mui/src/BoolField.tsx

```diff
@@ -0,0 +1,72 @@
+import Checkbox, { CheckboxProps } from '@mui/material/Checkbox';
+import FormControlLabel from '@mui/material/FormControlLabel';
+import FormGroup from '@mui/material/FormGroup';
+import FormLabel from '@mui/material/FormLabel';
+import Switch, { SwitchProps } from '@mui/material/Switch';
+import omit from 'lodash/omit';
+import React, { Ref } from 'react';
+import { FieldProps, connectField, filterDOMProps } from 'uniforms';
+
+import wrapField from './wrapField';
+
+export type BoolFieldProps = FieldProps<
+  boolean,
+  CheckboxProps | SwitchProps,
+  {
+    appearance?: 'checkbox' | 'switch';
+    fullWidth?: boolean;
+    helperText?: string;
+    legend?: string;
+    margin?: 'dense' | 'normal' | 'none';
+    transform?: (label: string) => string;
+  }
+>;
+
+function Bool(props: BoolFieldProps) {
+  const {
+    appearance,
+    disabled,
+    inputRef,
+    label,
+    legend,
+    name,
+    onChange,
+    readOnly,
+    transform,
+    value,
+  } = props;
+  const SelectionControl =
+    appearance === 'checkbox' || appearance === undefined ? Checkbox : Switch;
```

Comment:
Isn't `appearance === undefined` the most common case? If yes, I think it should be checked first. 
Also, isn't it quicker to check only if `appearance === 'switch'`?